tasks/workflow/worker.go

```diff
@@ -628,7 +628,11 @@ func (w *worker) runPurge(ctx context.Context, t *taskExecution) error {
 	if err != nil {
 		return err
 	}
-	return nil
+	// Just Publish an event that the deployment is successfully
+	// This event will stay into consul even if the deployment is actually purged...
+	// To prevent unexpected errors
+	_, err = events.PublishAndLogDeploymentStatusChange(ctx, t.cc.KV(), t.targetID, strings.ToLower(deployments.PURGED.String()))
```

Comment:
So for clients providing unique names to the applications they ask to deploy (like fastML), there will be few entries leftover in consul never removed for each purge of such application.
Could we store such purged deployment IDs in consul with a timestamp of the purge operation, and the final entire cleanup would be done either if a user deploys an application with the same name like today, or if the purge has occured since a given time ?

Reply:
Ok so lets define an eviction timeout after which remaining events & logs of a purged deployment will be deleted. Same as before if a deployment with the same ID is redeployed before the eviction timeout then those logs & events will also be deleted.